engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -209,6 +215,7 @@ public SlaveServer( String name, String hostname, String port, String username,
     this.master = master;
     initializeVariablesFrom( null );
     this.log = new LogChannel( this );
+    this.lock = new ReentrantReadWriteLock();
```

Comment:
As the first line of this constructor calls the zero-args constructor (which instances the lock), you shouldn't have this line.